packages/vscode-extension/src/project/launchConfigurationsManager.ts

```diff
@@ -131,6 +132,18 @@ export class LaunchConfigurationsManager implements Disposable {
     const savedLaunchConfig = workspaceState.get<LaunchConfiguration | undefined>(
       INITIAL_LAUNCH_CONFIGURATION_KEY
     );
+
+    if (savedLaunchConfig?.kind === LaunchConfigurationKind.Detected) {
+      const availableAppRoots = getAvailableApplicationRoots();
+      if (
+        availableAppRoots.some(
+          (availableAppRoot) => availableAppRoot.path === savedLaunchConfig.appRoot
+        )
+      ) {
+        return savedLaunchConfig;
+      }
+    }
+
     if (
       savedLaunchConfig &&
       this._launchConfigurations.find((config) => _.isEqual(config, savedLaunchConfig))
```

Comment:
We could probably check the kind here as well.


```suggestion
if (
savedLaunchConfig?.kind === LaunchConfigurationKind.Custom &&
this._launchConfigurations.find((config) => _.isEqual(config, savedLaunchConfig))
```